crypto/asn1/a_gentm.c

```diff
@@ -100,7 +119,7 @@ int asn1_generalizedtime_to_tm(struct tm *tm, const ASN1_GENERALIZEDTIME *d)
 
     if (a[o] == 'Z')
         o++;
-    else if ((a[o] == '+') || (a[o] == '-')) {
+    else if (!strict && ((a[o] == '+') || (a[o] == '-'))) {
```

Comment:
Oh, I guess you end up at the /* Missing time zone information. */ in that case, which is fine.